pkg/v1/remote/transport/bearer.go

```diff
@@ -144,3 +163,28 @@ func (bt *bearerTransport) refresh() error {
 	bt.bearer = &bearer
 	return nil
 }
+
+func (bt *bearerTransport) canonicalAddress(host string) (address string, err error) {
+	// The host may be any one of:
+	// - hostname
+	// - hostname:port
+	// - ipv4
+	// - ipv4:port
+	// - ipv6
+	// - [ipv6]:port
+	// As net.SplitHostPort returns an error if the host does not contain a port, we should only attempt
+	// to call it when we know that the address contains a port
+	if strings.Count(host, ":") == 1 || (strings.Count(host, ":") >= 2 && strings.Contains(host, "]")) {
+		hostname, port, err := net.SplitHostPort(host)
+		if err != nil {
+			return "", err
```

Comment:
I'm a bit worried that returning an error here might cause failures for things that used to work, e.g. if canonicalizing `in.Host` fails for some reason, but `in.URL.Host` would have succeeded, this would now fail.

What if we just use the original host in cases where we can't split it? Then we don't have to return an error (and don't have to handle it).

```go
hostname, port, err := net.SplitHostPort(host)
if err != nil {
  hostname = host
}
if port == "" {
  port = portMap[bt.scheme]
}
return net.JoinHostPort(hostname, port)
```

Reply:
Seems like a fair concern - I'll update this today.

Reply:
@jonjohnsonjr I believe @lw346's latest update addresses this concern. Can you take another look?

Reply:
Ah sorry I missed that this got updated.